components/builder-api/src/server/services/memcache.rs

```diff
@@ -206,6 +210,35 @@ impl MemcacheClient {
         ret
     }
 
+    pub fn get_origin_member_role(&mut self, origin: &str, account_id: u64) -> Option<String> {
+        trace!("Getting origin role membership for {} {} from memcached",
+               origin,
+               account_id);
+
+        let key = format!("member_role:{}/{}", origin, account_id);
+        let start_time = Instant::now();
+        let ret = self.get_string(&key);
+        let duration_millis = start_time.elapsed().as_millis();
+        trace!("Memcache get_origin_member_role time: {} ms",
+               duration_millis);
+        Histogram::MemcacheCallTime.set(duration_millis as f64);
+
+        ret
+    }
+
+    pub fn set_origin_member_role(&mut self, origin: &str, account_id: u64, role: &str) {
+        let key = format!("member_role:{}/{}", origin, account_id);
```

Comment:
Why doesn't this use member_role_ns_key?

Reply:
From what we can tell the hashing happens internal to memcache and is only necessary when using the API to pull out a value. Submitting a value this is the pattern that salim had already established so this is what we went with.

Reply:
Good catch! I originally added a comment here that was way off base - i had been confused about which function that was. Corrected!